lib/src/configuration.dart

```diff
@@ -222,16 +215,20 @@ class FlexibleSyncConfiguration extends Configuration {
   final User user;
 
   SessionStopPolicy _sessionStopPolicy = SessionStopPolicy.afterChangesUploaded;
-  
-  /// Called when a [SessionError] occurs for the synchronized Realm.
-  final SessionErrorHandler? sessionErrorHandler;
+
+  /// Called when a [SyncError] occurs for this synchronized [Realm].
+  final SyncErrorHandler? syncErrorHandler;
+
+  /// Called when a [SyncClientResetError] occurs for this synchronized [Realm]
+  final SyncClientResetErrorHandler? syncClientResetErrorHandler;
```

Comment:
This is possible only if we move it to required in the flexible sync config? Are you suggesting that?

Reply:
we have manual reset only currently no? how do you suggest to supply a default one?

Reply:
We can construct one that just logs and tells people to do something themselves:

```dart
final defaultHandler = ManualResetHandler((_) {
  Realm.logger.log(LogLevel.Error, "A client reset error occurred but no handler was supplied. "
    "Synchronization is now paused and will resume automatically once the app is restarted and "
    "the server data is redownloaded. Any unsynchronized changes the client has made or will "
    "make will be lost. To handle that scenario, pass in a non-null value to 
    "`syncClientResetErrorHandler` when constructing `Configuration.flexibleSync`");
});
```

It doesn't do anything to handle the error but instructs the user what they need to do since the Core error message is going to be overly generic and not geared toward the Dart SDK API.

Reply:
I think this is not ideal. Why not make it to require. I am for moving it as a required field right now later when we have more types we can relax the constraint.  
Also this kind of message should be an exception. This is really hiding the issue until later when the person might have shipped the app. So we better be clear we require this field. ClientResets can happen regularly. 
Also our SyncErrorHandler should be a required field. Thus we have a chance to really raise any sync issue to the developer. They can choose to handle it as printing or something else.

Reply:
since there is no time for better I will implement what you suggested.